components/hab/src/main.rs

```diff
@@ -1490,6 +1501,51 @@ where
     return Ok(());
 }
 
+/// Check if we have a launcher/supervisor running out of this habitat root.
+/// If the launcher PID file exists then the supervisor is up and running
+fn launcher_is_running(fs_root_path: &Path) -> bool {
+    let launcher_root = launcher_root_path(Some(fs_root_path));
+    let pid_file_path = launcher_root.join("PID");
+
+    pid_file_path.is_file()
+}
+
+fn supervisor_services() -> Result<Option<Vec<PackageIdent>>> {
+    if !launcher_is_running(&*FS_ROOT) {
+        return Ok(None);
+    }
+
+    let cfg = config::load()?;
+    let secret_key = ctl_secret_key(&cfg)?;
+    let sup_addr = protocol::ctl::default_addr();
+    let msg = protocol::ctl::SvcStatus::default();
+
+    let mut out: Vec<PackageIdent> = vec![];
+    SrvClient::connect(&sup_addr, &secret_key)
+        .and_then(|conn| {
+            conn.call(msg).for_each(|reply| {
+                match reply.message_id() {
+                    "ServiceStatus" => {
+                        let m = reply.parse::<protocol::types::ServiceStatus>().unwrap();
```

Comment:
Are these `unwraps` logically guaranteed to be `Ok`? If not, we should probably handle them rather than panic.

Reply:
Lemme check.  I copied from the doc example so it might not be bulletproof.

Reply:
Looks like they are guaranteed to be Ok.  All our other sup communication code in `main.rs` follows the same pattern of using `unwrap()`.

Reply:
We've been wrong before https://github.com/habitat-sh/habitat/issues/5148 https://github.com/habitat-sh/habitat/issues/5151, so I'm not sure the fact that existing code uses that pattern is totally convincing. If you can point to an earlier point in the code that would handle `Err` values before they get here, that's different.

Reply:
Indeed.  Diving through the marshalling code it looks like it's not handled properly in any of these calls.  Will fix.